### PR TITLE
[front] - feat(vault): vault creation

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at plugin the `CreateVaultModal` to the "Create Vault" button of the `VaultSidebarMenu`

## Risk

Few

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/components/vaults/CreateVaultModal.tsx [hunk 1]
```diff
@@ -28,11 +34,10 @@ type RowData = {
 type Info = CellContext<RowData, unknown>;
 
 interface CreateVaultModalProps {
-  owner: WorkspaceType;
+  owner: WorkspaceType | LightWorkspaceType;
```
Comment (flvndvd) [inline front/components/vaults/CreateVaultModal.tsx:37]:
```suggestion
  owner: LightWorkspaceType;
```

File: front/components/vaults/CreateVaultModal.tsx [hunk 2]
```diff
@@ -28,11 +34,10 @@ type RowData = {
...
   isOpen: boolean;
   onClose: () => void;
-  onSave: () => void;
-  allUsers: UserType[];
+  onSave?: () => void;
```
Comment (flvndvd) [inline front/components/vaults/CreateVaultModal.tsx:40]: Why is this one optional?

File: front/lib/swr.ts [hunk 3]
```diff
@@ -320,7 +321,7 @@ export function useDataSources(
   };
 }
 
-export function useMembers(owner: WorkspaceType) {
+export function useMembers(owner: WorkspaceType | LightWorkspaceType) {
```
Comment (flvndvd) [inline front/lib/swr.ts:324]: Type inheritance
```suggestion
export function useMembers(owner: LightWorkspaceType) {
```

File: front/components/vaults/VaultSideBarMenu.tsx [hunk 4]
```diff
@@ -76,6 +79,12 @@ export default function VaultSideBarMenu({ owner }: VaultSideBarMenuProps) {
           );
         })}
       </Item.List>
+      <CreateVaultModal
```
Comment (flvndvd) [inline front/components/vaults/VaultSideBarMenu.tsx:82]: I'm not convince that this should live here. IMO it should be in the VaultLayout component and watches the router.

File: front/lib/swr.ts (hunk 3 above)
Comment (JulesBelveze) [reply front/lib/swr.ts:324]: `useMembers` is used in the workspace members view and owner is of type `WorkspaceType` there

Comment (flvndvd) [reply front/lib/swr.ts:324]: Yep but IIRC, `WorkspaceType` extends `LightWorkspaceType` so you should be good to go with `LightWorkspaceType`

**Review by flvndvd — APPROVED** [review]

File: front/components/vaults/VaultSideBarMenu.tsx (hunk 4 above)
Comment (flvndvd) [reply front/components/vaults/VaultSideBarMenu.tsx]: LGTM

**Pull request merged**